app/assets/javascripts/discourse/app/components/modal/sidebar-section-form.js

```diff
@@ -232,22 +232,22 @@ class SectionLink {
   }
 }
 
-export default Controller.extend(ModalFunctionality, {
-  dialog: service(),
-  router: service(),
+export default class SidebarSectionForm extends Component {
+  @service dialog;
+  @service router;
 
-  onShow() {
-    this.setProperties({
-      flashText: null,
-      flashClass: null,
-    });
+  @tracked flash;
+  @tracked flashType;
+
+  init() {
     this.nextObjectId = 0;
     this.model = this.initModel();
-  },
+    super.init(...arguments);
+  }
```

Comment:
nextObjectId could be initialized as a class field like:

```
nextObjectId = 0;
```

It's a bit strange that `initModal` looks at the incoming `modal` argument and replaces it with a transformed version. That's a bit of a code smell, and isn't even possible in Glimmer components.

I think the ideal solution would be if `initModel` could be converted to a getter like

```
@cached
get transformedModal(){
```

But I understand if you don't want to go down that refactoring rabbit hole right now - this is not a blocking comment.